internal/pkg/aws/identity/identity.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package identity wraps AWS Security Token Service (STS) API functionality.
+package identity
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/aws/aws-sdk-go/service/sts/stsiface"
+)
+
+// Service wraps the internal sts client.
+type Service struct {
+	sts stsiface.STSAPI
+}
+
+// New returns a Service configured with the input session.
+func New(s *session.Session) Service {
+	return Service{
+		sts: sts.New(s),
+	}
+}
+
+// Caller holds information about a calling entity.
+type Caller struct {
+	Arn     string
```

Comment:
nit: I think the idiomatic way is "ARN" https://github.com/golang/go/wiki/CodeReviewComments#initialisms

Reply:
I haven't seen consistency regarding this. CloudFormation resource properties and references use `Arn`. Most documentation seems to use `ARN`. Some existing references in the codebase are `Arn`. I'm down for whichever, but we should be consistent.

Reply:
I'd lean towards doing what the Go style guide recommends instead of other 3rd party packages :P